### PR TITLE
feat: Add specific component tags for avatar, text-area and button for better theming support

### LINKED ISSUE #28: Define specific tag names for avatar, text area and button
As pointed out by @tomivirkki. When using an external component in another component, the convention is to define a new tag name for it.

In `vaadin-message`:
- Define `vaadin-message-avatar` and use it instead of `vaadin-avatar`.

In `vaadin-message-input`:
- Define `vaadin-message-input-text-area` and use it instead of `vaadin-text-area`.
- Define `vaadin-message-input-button` and use it instead of `vaadin-button`.

[See for example the `<vaadin-text-field>` inside `<vaadin-date-picker>`](
https://github.com/vaadin/vaadin-date-picker/blob/ba6c8be0bc1050d87f05326ae00502d609daa2ff/src/vaadin-date-picker-text-field.js#L41-L47).

### PR DESCRIPTION
**Pull request by Peppe**
Resolves #28

### CODE REVIEW

File: src/vaadin-message-input-text-area.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ *
+ * See [ThemableMixin – how to apply styles for shadow parts](https://github.com/vaadin/vaadin-themable-mixin#readme)
+ *
+ * @extends TextAreaElement
+ */
+class MessageInputTextAreaElement extends TextAreaElement {
+  static get is() {
+    return 'vaadin-message-input-text-area';
+  }
+}
```
Comment (web-padawan) [inline src/vaadin-message-input-text-area.js:37]: This component could contain `ready` block from `vaadin-message-input` and configure some defaults:

https://github.com/vaadin/vaadin-messages/blob/beef43f8c81ca076a16f4295571dc185a41d642f/src/vaadin-message-input.js#L112-L130

**Review by web-padawan — CHANGES_REQUESTED** [review]
Visual tests are failing, please check if some CSS is not correctly applied.

**Comment by Peppe** [comment]
I have a problem.
Linting started complaining about `vaadin-message-avatar.js:13`:
```css
--vaadin-avatar-outline-width: 0px;
```
It is a zero value and zero values should not have a unit. I changed the row to 
```css
--vaadin-avatar-outline-width: 0;
```
but that actually broke the implementation. Margin-right of the avatar was not anymore there. This is because it is used in a calc in `vaadin-message-avatar-styles.js:8`:
```css
:host {
  margin-right: calc(var(--lumo-space-m) - var(--vaadin-avatar-outline-width));
  margin-top: calc(var(--lumo-space-s) - var(--vaadin-avatar-outline-width));
}
```
that evaluates with resolved variables to `margin-right: calc(1rem - 0);`, and I guess you have to have units on everything that goes into a calc.

So now I'm unsure how to have to satisfy the linter correctly.

File: src/vaadin-message-input-text-area.js (hunk 1 above)
Comment (Peppe) [reply src/vaadin-message-input-text-area.js:37]: `this.__submit()` is part of the `vaadin-message-input.js` file. Do I need to dispatch an event in `vaadin-message-input-text-area`, then catch it in `vaadin-message-input`, and call `__submit();`?

Comment (Peppe) [reply src/vaadin-message-input-text-area.js:66]: I did just that. Done.

File: src/vaadin-message-avatar.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { AvatarElement } from '@vaadin/vaadin-avatar/src/vaadin-avatar.js';
+import { registerStyles, css } from '@vaadin/vaadin-themable-mixin/register-styles.js';
+
+registerStyles(
+  'vaadin-message-avatar',
+  css`
+    :host {
+      --vaadin-avatar-outline-width: 0px;
```
Comment (web-padawan) [inline src/vaadin-message-avatar.js:13]: @Peppe You could use an ignore comment to get rid of Stylelint error:

```css
--vaadin-avatar-outline-width: 0px;
 /* stylelint-disable-line declaration-no-important */
```

I will update our Stylelint config later on.

**Review by web-padawan — APPROVED** [review]

**Comment by web-padawan** [comment]
Tested in iOS Simulator locally and the tests are passing. Probably it was SauceLabs issue.

**Pull request merged**